core/rawdb/chain_freezer.go

```diff
@@ -118,24 +117,23 @@ func (f *chainFreezer) readFinalizedNumber(db ethdb.KeyValueReader) uint64 {
 	return *number
 }
 
-// freezeThreshold returns the threshold for chain freezing. It's determined
-// by formula: max(finality, HEAD-params.FullImmutabilityThreshold).
+// freezeThreshold returns the threshold for chain freezing.
```

Comment:
It seems like the original function is actually good. It's 100% o.k to have a threshold that won't be frozen even if finalized.
Can we remove the change to geth entirely?